### PR TITLE
Improvements to SAW Lab Proofs and Makefiles

### PR DESCRIPTION
**Pull request by djmorel**
# Features
- Updates the SAW labs based on @WeeknightMVP 's comments about [relative paths for proofs](https://github.com/weaversa/cryptol-course/pull/205#issuecomment-1152993146) and [specifying targets for Makefiles](https://github.com/weaversa/cryptol-course/pull/205#issuecomment-1152995699).
  - Python proof scripts can now be called anywhere!
- Updates the python-saw CI to run SAW labs in a `make prove -C` format without the need to change directories via `cd`
- Updated the SAW lab Makefiles to be identical except for their `LAB` variables
- Updated the SAW.md documentation to reflect these changes

### CODE REVIEW

**Review by weaversa — APPROVED** [review]
Great work @djmorel! This makes the dual tasks of development and CI integration work together seamlessly, and provides some very nice examples others can build off of.

I'll approve but wait for @WeeknightMVP to have a say before merging, seeing as he submitted the original issue.

**Review by WeeknightMVP — COMMENTED** [review]
Externalizing a `LAB` variable in `Makefile`s is nice, as it establishes a consistent process for SAW Remote API builds and proofs.  So consistent that their content after `LAB=name` is the same.  (Except for `rotl`, where `rotl3.c` would just need to be renamed `rotl.c`.  Enumerating intermediate files and declaring the last one as the representative makes sense.)  This suggests that each lab's `Makefile` could be [`DRY`]ed to something like:

```make
LAB=name
# any other variables or rules specific to this lab

include "../prove.mk"  # Nested labs would require more `..`s
```

```make
# labs/SAW/proof.mk

# Note: Makefile assumes user already started the SAW remote API
# $ start-saw-remote-api

# Variables to hold relative file paths
SRC=src
PROOF=proof
SPECS=specs
ARTIFACTS=artifacts

all: build

build: $(ARTIFACTS)/$(LAB).bc

prove: build
	python3 $(PROOF)/$(LAB).py

clean:
	rm -rf $(ARTIFACTS)

.PHONY: all build prove clean

$(ARTIFACTS)/$(LAB).bc: $(SRC)/$(LAB).c | $(ARTIFACTS)
	clang -c -g -emit-llvm -o $@ $<

$(ARTIFACTS):
	mkdir -p $(ARTIFACTS)
```

From here, a reasonable next step (in future PRs) would be to leverage `Makefile`s with `-include` directives and/or shell scripts to build and run all (or all specified) SAW Remote API proof scripts locally (and call it from the CI configuration).  However that proceeds, we should decide whether to combine the tutorial with the other `saw-proofs` or...do something else with those.

We should also consider whether to have proofs generate log files, and whether `all` should just build bitcode files or also run/log proofs, e.g.:

```make
prove: $(ARTIFACTS)/$(LAB).log

$(ARTIFACTS)/$(LAB).log: $(PROOF)/$(LAB).py $(ARTIFACTS)/$(LAB).bc
	python3 $< > $@
```

**Comment by WeeknightMVP** [comment]
From here, a reasonable next step (in future PRs) would be to leverage `Makefile`s with `-include` directives and/or shell scripts to build and run all (or all specified) SAW Remote API proof scripts locally (and call it from the CI configuration).  However that proceeds, we should decide whether to combine the tutorial with the other `saw-proofs` or...do something else with those.

**Comment by WeeknightMVP** [comment]
We should also consider whether to have proofs generate log files, and whether `all` should just build bitcode files or also run/log proofs, e.g.:

```make
prove: $(ARTIFACTS)/$(LAB).log

$(ARTIFACTS)/$(LAB).log: $(PROOF)/$(LAB).py $(ARTIFACTS)/$(LAB).bc
	python3 $< > $@
```

**Comment by WeeknightMVP** [comment]
Is [`labs/SAW/Game/DLC/Makefile`](https://github.com/weaversa/cryptol-course/blob/5b39ae9e9617f8910cb92d592196634086afff75/labs/SAW/Game/DLC/Makefile) meant to overwrite `artifacts/SAW/Game`, or create its own directory `artifacts/SAW/Game/DLC`?

**Comment by djmorel** [comment]
I'm going to temporarily restore the `relative-proofs` branch to make the rotl lab use the new Makefile format. I'll also take the opportunity to update `SAW.md` so to reflect the naming changes @WeeknightMVP proposed.

**Pull request merged**
